frontend/app_flowy/packages/appflowy_editor/lib/src/operation/transaction_builder.dart

```diff
@@ -114,21 +115,17 @@ class TransactionBuilder {
 
   /// Inserts content at a specified index.
   /// Optionally, you may specify formatting attributes that are applied to the inserted string.
-  /// By default, the formatting attributes before the insert position will be used.
+  /// When no formatting attributes specified, the formating attributes before the insert position will be used if they don't have defaultFormatting flag set
+  /// When defaultFormatting flag is set before the insert position, it will be cleared.
+  /// When insert position is within a text having defaultFormatting flag set, the flag will be ignored and clear (formatting attributes of the text will be applied)
   insertText(
     TextNode node,
     int index,
     String content, {
     Attributes? attributes,
   }) {
-    var newAttributes = attributes;
-    if (index != 0 && attributes == null) {
-      newAttributes =
-          node.delta.slice(max(index - 1, 0), index).first.attributes;
-      if (newAttributes != null) {
-        newAttributes = Attributes.from(newAttributes);
-      }
-    }
+    final newAttributes = attributes ?? _getAttributesAt(node, index);
```

Comment:
I don't think it's the best solution, it's good for now and it works. Maybe we need to introduce a new object to sealing formatting of a previous content. So far this works okay, but I'm open do do it in a different way.